.github/workflows/main.yml

```diff
@@ -70,8 +70,15 @@ jobs:
           python --version
           cmake --version
 
+      - name: Show versions g++
+        if: (matrix.os == 'ubuntu-latest' || matrix.os == 'ubuntu-24.04')
+        run: |
+          sudo apt install g++-11
+          sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 12
+          g++ --version          
+
       - name: Install external protobuf - Linux
-        if: matrix.protobuf_type == 'External' && matrix.os == 'ubuntu-latest'
+        if: matrix.protobuf_type == 'External' && (matrix.os == 'ubuntu-latest' || matrix.os == 'ubuntu-24.04')
```

Comment:
Startswith ubuntu?

Reply:
yes, thank you

Reply:
I think you may use https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#startswith to simplify this expression

Reply:
adapted